library/Zend/Mail/Message.php

```diff
@@ -36,6 +36,44 @@ class Message
     protected $encoding = 'ASCII';
 
     /**
+     * Constructor.
+     *
+     * @param null|array|Traversable $options (Optional) Options to set
+     * @param array|Traversable $options
+     */
+    public function __construct($options = null)
```

Comment:
@Tauop I think what @Ocramius is getting at is that you're introducing a new API into `Zend\Mail\Message` by introducing the constructor (which did not exist before). I'm looking at it, and I'm of the opinion that it may make more sense to introduce a factory method instead, either in this class, or as a separate class. Usage would then be something like:

```php
$mail = MailFactory::getInstance($options);
```

@Ocramius I can see why the new API would be useful, though, in the particular situation that is proposed in this PR. Since the various message members are only known based on the configuration provided, and the specific members could vary based on invocation, having a factory becomes necessary, as `Zend\Mail\Message` is a value object.